### PR TITLE
fix getting all available output for interactive commands + fix logging end of output when no match was found in `run_shell_cmd`

### PR DESCRIPTION
**Pull request by boegel**
bug fix, required for installing Maple (cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/3286)

### CODE REVIEW

File: easybuild/tools/run.py [hunk 1]
```diff
@@ -425,12 +425,20 @@ def to_cmd_str(cmd):
...
+            more_stdout = True
+            while more_stdout:
+                more_stdout = proc.stdout.read(read_size) or b''
+                stdout += more_stdout
 
             # note: we assume that there won't be any questions in stderr output
             if split_stderr:
-                stderr += proc.stderr.read1(read_size) or b''
+                more_stderr = True
+                while more_stdout:
```
Comment (branfosj) [inline easybuild/tools/run.py:439]:
```suggestion
                while more_stderr:
```

Comment (boegel) [reply easybuild/tools/run.py]: fixed in c090f49056eccf165e3a493c9ad0c8b19d210817

File: easybuild/tools/run.py [hunk 2]
```diff
@@ -425,18 +425,26 @@ def to_cmd_str(cmd):
...
+                while more_stderr:
+                    more_stderr = proc.stderr.read(read_size) or b''
+                    stderr += more_stderr
 
             if qa_patterns:
                 if _answer_question(stdout, proc, qa_patterns, qa_wait_patterns):
                     time_no_match = 0
                 else:
-                    _log.debug(f"No match found in question/wait patterns at end of stdout: {stdout[:1000]}")
+                    _log.debug(f"No match found in question/wait patterns at end of stdout: {stdout[:-1000]}")
```
Comment (boegel) [inline easybuild/tools/run.py:447]: This needs more work, don't merge yet please...

Comment (boegel) [reply easybuild/tools/run.py]: fixed in 0e156571d

**Review by branfosj — APPROVED** [review]

**Pull request merged**